astropy/io/ascii/tests/test_write_read.py

```diff
@@ -0,0 +1,153 @@
+    try:
+        if debug:
+            print(test_def['kwargs'])
+            print(test_read_def['kwargs'])
+        ascii.write(table, in_out, fast_writer=fast_writer,
+                    **test_def['kwargs'])
+        foo = ascii.read(in_out, **test_read_def['kwargs'])
+    except ValueError as e:  # if format doesn't have a fast writer, ignore
+        if 'not in the list of formats with fast writers' not in str(e):
+            raise e
+        return
+
+    if debug:
+        print('Expected:\n%s' % test_def['out'])
+        print('Actual:\n%s' % in_out.getvalue())
+        print('Read:\n%s' % str(foo))
+
+    assert [x.strip() for x in in_out.getvalue().strip().splitlines()] == \
+           [x.strip() for x in test_def['out'].strip().splitlines()]
+
+
+def check_read_write_table_via_table(test_def, table, fast_writer):
+    """Check reading and writing using the astropy.table.Table interface."""
+    in_out = StringIO()
+
+    test_write_def = copy.deepcopy(test_def)
+    # Table would like 'format' text string instead of 'Writer' and 'Reader'.
+    if 'Writer' in test_def['kwargs']:
+        format_name = 'ascii.{0}'.format(
+            test_write_def['kwargs']['Writer']._format_name)
+        test_write_def['kwargs']['format'] = format_name
+        del test_write_def['kwargs']['Writer']
+
+    test_read_def = create_reader_kwargs_from_writer_kwargs(test_write_def)
+
+    try:
+        table.write(in_out, fast_writer=fast_writer,
+                    **test_write_def['kwargs'])
+        foo = Table.read(in_out, **test_read_def['kwargs'])
+    except ValueError as e:  # if format doesn't have a fast writer, ignore
+        if 'not in the list of formats with fast writers' not in str(e):
+            raise e
+        return
+
+    if debug:
+        print('Expected:\n%s' % test_def['out'])
+        print('Actual:\n%s' % in_out.getvalue())
+        print('Read:\n%s' % str(foo))
+
+    assert [x.strip() for x in in_out.getvalue().strip().splitlines()] == \
+           [x.strip() for x in test_def['out'].strip().splitlines()]
+
+
+def writer_or_format_is_html(def_kwargs):
+    """Identify a test_def that will write/read HTML.
+
+    If 'Writer': ascii.HTML or 'format': 'ascii.html'
+    then return False
+    Otherwise return True
+    >>> print(writer_or_format_is_html({'Writer': ascii.HTML}))
+    True
+    >>> print(writer_or_format_is_html({'format': 'ascii.html'}))
+    True
+    >>> print(writer_or_format_is_html({'Writer': ascii.Csv}))
+    False
+    """
+    if 'Writer' in def_kwargs and def_kwargs['Writer'] == ascii.HTML:
+        return True
+    if 'format' in def_kwargs and def_kwargs['format'] == 'ascii.html':
+        return True
+
+    return False
+
+
+check_functions = \
+    (check_read_write_table_via_ascii, check_read_write_table_via_table)
+
+
+@pytest.mark.parametrize("test_def", test_defs)
+@pytest.mark.parametrize("check_function", check_functions)
+@pytest.mark.parametrize("fast_writer", [True, False])
```

Comment:
definitely a good use of parameterize here to get a lot of cases covered with very little new code.